docs/guide/vite.md

```diff
@@ -0,0 +1,383 @@
+
+## Installation
+Moving forward, all of our starter kits will have the Vite build option available. If you are creating a project from scratch without a starter kit, or want to enable Vite builds in an existing project you can install the package from npm.
+
+```bash
+npm install @apostrophecms/vite
+```
+
+The package then needs to be added to the `app.js` file:
+
+<AposCodeBlock>
+
+```javascript
+require('apostrophe')({
+  shortName: 'my-project',
+  modules: {
+    '@apostrophecms/vite': {},
+  }
+});
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+
+</AposCodeBlock>
+
+## Core Features and Configuration
+
+The Vite bundler for ApostropheCMS comes with sensible defaults while remaining highly configurable. Hot Module Replacement (HMR) is enabled out of the box for project UI code, allowing you to see your changes instantly without a full page refresh.
+
+### Hot Module Replacement
+
+HMR can be configured in three modes using the `@apostrophecms/asset` options:
+
+<AposCodeBlock>
+
+```javascript
+// in app.js
+require('apostrophe')({
+  shortName: 'my-project',
+  modules: {
+    '@apostrophecms/vite': {},
+    '@apostrophecms/asset': {
+      options: {
+        // Choose one of these HMR options:
+        hmr: 'public',  // Default - enables HMR for project UI
+        hmr: 'apos',    // Enables HMR for Admin UI
+        hmr: false      // Disables HMR completely
+      }
+    }
+  }
+});
+```
+ <template v-slot:caption>
+    app.js
+  </template>
+
+</AposCodeBlock>
+
+::: info
+If you use the `hmr: 'apos'` option, you don't need to pass the `APOS_DEV` flag when editing components of the Admin UI. The Vite build will selectively update altered code leading to a better development experience than `APOS_DEV=1`, which will rebuild the entirety of the Admin UI code.
+:::
+
+### WebSocket Configuration
+
+By default, the HMR WebSocket server runs on your ApostropheCMS server port, typically port 3000. For cases where you need a separate port (like running behind certain proxy configurations), you can configure a custom port:
+
+<AposCodeBlock>
+
+```javascript
+'@apostrophecms/asset': {
+  options: {
+    hmrPort: 3001  // Sets a custom WebSocket server port
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+### Source Maps in Production
+
+For debugging production builds, you can enable source maps to see the original source code in your browser's developer tools:
+
+<AposCodeBlock>
+
+```javascript
+'@apostrophecms/asset': {
+  options: {
+    productionSourceMaps: true  // Exposes source maps in production
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+::: tip
+Source maps slightly increase your build size but are invaluable for debugging production issues. Consider your specific needs when enabling this option.
```

Comment:
The production source maps are not inline, but referenced (not increasing the actual loaded JS) - the browser should load them separately.

Reply:
Tip removed. I thought I read this in the Vite documentation - can't find it now.